File: test/Test_PterodactylEgg.cpp
#include "../include/PterodactylEgg.cpp"
#include <gtest/gtest.h>
#include <string>

using namespace std;

TEST(PterodactylEggTest, AllFunctionalityofPteradactylEgg){
    PterodactylEgg testPE(500, "oldie");
    ASSERT_EQ("oldie", testPE.getProducer());
    ASSERT_EQ(PTERODACTYL_EGG, testPE.getObjectType());
    ASSERT_EQ(FARMPRODUCT, testPE.getProductType());
    ASSERT_EQ(500, testPE.getPrice());
}

File: Java/src/test/java/engisubs/gameobject/product/farmproduct/egg/TestPterodactylEgg.java
package engisubs.gameobject.product.farmproduct.egg;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import engisubs.gameobject.GameObject.GameObjectType;
import engisubs.gameobject.product.Product.ProductType;
public class TestPterodactylEgg{
    private PterodactylEgg test;
    @Before
    public void initObject(){
        test = new PterodactylEgg(25, "Name");
    }

    @Test
    public void instantiationTest(){
        Assert.assertEquals("Name", test.getProducer());
        Assert.assertEquals(GameObjectType.PTERODACTYL_EGG, test.getObjectType());
        Assert.assertEquals(ProductType.FARMPRODUCT, test.getProductType());
        Assert.assertEquals(25, test.getPrice());
    }

}